### ISSUE TITLE
How do we restore training from the last checkpoint?

### ISSUE THREAD

**Issue by cxfire16** [MarvinTeichmann/KittiSeg#156]
Greetings!

I forgot to thank you on my last thread; Thank you for sharing your code! I've been doing many experiments and I've had a lot of fun in doing so. On my most recent experiment, I experienced a sudden power interruption and my UPS unit wasn't able to sustain the sudden draw of a system bearing 1080ti at full load and the training was lost(or is it?). This made me realize if it was indeed possible to resume training on a saved checkpoint. I've attempted to do it through the tensorvision as written on the readme but to no avail I was able to restore the training. I believe I must be missing something as when I try to use the "continue.py" script from tensorvision, it would just end program after a few initialization of the Environment variables as described by the log below. I would be more than happy to interact and to soon be able to contribute to this project. Thanks and have a great day!

(tensorgpu-1) trainer@trainer-1:/trainer/venvs/tensorgpu-1/data/multiclass-fcn/walldetector$ python continue.py --logdir=RUNS/test_11_restore
2018-01-29 16:31:39,247 INFO No environment variable 'TV_PLUGIN_DIR' found. Set to '/home/chuck/tv-plugins'.
2018-01-29 16:31:39,247 INFO No environment variable 'TV_STEP_SHOW' found. Set to '50'.
2018-01-29 16:31:39,247 INFO No environment variable 'TV_STEP_EVAL' found. Set to '250'.
2018-01-29 16:31:39,247 INFO No environment variable 'TV_STEP_WRITE' found. Set to '1000'.
2018-01-29 16:31:39,247 INFO No environment variable 'TV_MAX_KEEP' found. Set to '10'.
2018-01-29 16:31:39,247 INFO No environment variable 'TV_STEP_STR' found. Set to 'Step {step}/{total_steps}: loss = {loss_value:.2f}; lr = {lr_value:.2e}; {sec_per_batch:.3f} sec (per Batch); {examples_per_sec:.1f} imgs/sec'.
(tensorgpu-1) trainer@trainer-1:/trainer/venvs/tensorgpu-1/data/multiclass-fcn/walldetector$

I extracted the continue.py from the tensorvision folders and added  sys.path.insert(1, 'submodules') for it to be able to import the rest of tensorvision, quite crude but I needed to try. It's quite frustrating when you're training for hours and a sudden power out would halt the training so I really wish to learn on how to deal with such occurrences.

**Comment by cxfire16** [comment]
Solved by deriving a script that exposes the RUNS directory to the tensorvision's continue_training function under the train module. Thanks!
